### DISCUSSION TITLE
How does OpenFAST consider the effects of blade vibrations on aerodynamic forces?

### DISCUSSION THREAD

**Discussion by ietqlw** [OpenFAST/openfast#1123] [Q&A]
Dear all,

@jjonkman 

Vibrations of flexible blades are likely to affect aerodynamic forces. I guess that OpenFAST has already taken the effect into account. Could you please shear me with the details about it. 

How about the blade displacements, velocities, and accelerations on aerodynamic forces?

For example, the blade velocities might (1) not affect the aerodynamic forces, (2) affect the relative wind speed that the blades experience, (3) affect the absolute inflow wind condition besides (2). Which kind of mechanism is applied in OpenFAST?

Liangwen Qi
Chinese Academy of Science

**Comment by jjonkman** [comment]
Dear @ietqlw,

Yes, OpenFAST considers aero-elastic effects associated with the coupling between the aerodynamic forcing and structural response.  I would refer to the various theory bases of the structural modules (ElastoDyn, BeamDyn), aerodynamic module (AeroDyn), and coupling (glue code) for detailed information about how these are implemented.

But basically, at each glue code time step, the motion of the structural analysis nodes is computed by the structural module and used as input to the aerodynamics module, from which the aerodynamic loads are calculated and output and used as input to the structural module.  The structural displacement effects where in the wind field the blade is and its orientation relative to the wind and a relative velocity between the ambient wind and structural velocity is used in the aerodynamic calculations (akin to item (2) in your list, if I interpret correctly).

Best regards,
